helix-view/src/document.rs

```diff
@@ -1122,13 +1130,6 @@ impl Document {
         self.syntax.as_ref()
     }
 
```

Comment:
With this config removed `tab-width` is not updated on `:config-reload` anymore. I actually think that we should keep this function around and not make `IndentStyle` a struct at all. 

Rather we should add a second function to the document to compute the `indent_width`:
``` suggestion
     /// Tab size in columns.
    pub fn tab_width(&self) -> usize {
        self.language_config()
            .and_then(|config| config.indent.as_ref())
            .map_or(4, |config| config.tab_width) // fallback to 4 columns
     }

    /// Indentation size in columns
    pub fn indent_width(&self) -> usize {
        match self.indent_style{
            IndentStyle::Tab => self.tab_width(),
            IndentStyle::Spaces(width) => width as usize
         }
    }
```